### PR TITLE
Add new features

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: contracts/ARTHX/ARTHX.sol [hunk 1]
```diff
@@ -158,4 +176,25 @@ contract ARTHShares is AnyswapV4Token, IARTHX {
...
+
+    function _transfer(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) internal virtual override notPaused onlyNonBlacklisted(sender) {
+        // Check if tax mode is on, and if its is on tax the tx and transfer the
+        // remaining amount to `recipient`.
+        if (address(_taxController) != address(0))
+            amount = _taxController.chargeTax(_msgSender(), amount);
```
Comment (senamakel) [inline contracts/ARTHX/ARTHX.sol]: this is going to cost way too much gas; instead what we should do is send the tokens to another contract and take things from there

File: contracts/Arth/Arth.sol [hunk 2]
```diff
@@ -12,20 +15,46 @@ import {AnyswapV4Token} from '../ERC20/AnyswapV4Token.sol';
...
     /// @notice This is to help with establishing the Uniswap pools, as they need liquidity.
-    uint256 public constant override genesisSupply = 22000000e18; // 22M ARTH (testnet) & 5k (Mainnet).
+    uint256 public constant override genesisSupply = 22_000_000 ether; // 22M ARTH (testnet) & 5k (Mainnet).
+
+    address public troveManagerAddress;
+    address public stabilityPoolAddress;
+    address public borrowerOperationsAddress;
 
-    mapping(address => bool) public pools;
+    mapping(address => bool) public override pools;
```
Comment (senamakel) [inline contracts/Arth/Arth.sol:49]: no point having an override over here

File: contracts/Arth/ArthController.sol [hunk 3]
```diff
@@ -149,6 +156,54 @@ contract ArthController is AccessControl, IARTHController {
      * External.
      */
 
+    function toggleUseGlobalCRForMint(bool flag)
+        external
+        override
+        onlyByOwnerGovernanceOrPool
+    {
+        useGlobalCRForMint = flag;
```
Comment (senamakel) [inline contracts/Arth/ArthController.sol:173]: need to add events for these functions

File: contracts/Arth/Arth.sol (hunk 2 above)
Comment (ghost) [reply contracts/Arth/Arth.sol:49]: This is being used to verify if an address/contract is set as pool in the ARTHX. https://github.com/MahaDAO/arthcoin-v2/blob/add-new-features/contracts/ARTHX/ARTHX.sol#L59

File: contracts/ARTHX/ARTHX.sol (hunk 1 above)
Comment (ghost) [reply contracts/ARTHX/ARTHX.sol]: Understood, made the necessary change, kindly check again.

File: contracts/ARTHX/ARTHX.sol [hunk 4]
```diff
@@ -158,4 +186,27 @@ contract ARTHShares is AnyswapV4Token, IARTHX {
...
+
+    function _transfer(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) internal virtual override notPaused onlyNonBlacklisted(sender) {
+        if (taxPercent > 0 && address(_taxController) != address(0)) {
+            uint256 taxAmount = amount.mul(taxPercent).div(100);
+            super._transfer(sender, address(_taxController), taxAmount);
+            _taxController.chargeTax(); // Should we call this here? Or have a call function in controller, which at once does this?
```
Comment (senamakel) [inline contracts/ARTHX/ARTHX.sol:198]: remove line 198 to save on gas

**Review by senamakel — APPROVED** [review]

**Pull request merged**